subsys/bluetooth/mesh/sensor_types.c

```diff
@@ -521,13 +541,111 @@ static bool percentage_delta_check(const struct bt_mesh_sensor_value *delta,
 	return (diff / prev) > (percentage_float);
 }
 
+static bool scalar_unknown_raw(const struct bt_mesh_sensor_format *format,
+			       int64_t *val)
+{
+	const struct scalar_repr *repr = format->user_data;
+
+	if (repr->flags & HAS_UNDEFINED) {
+		*val = scalar_type_max(format);
+	} else if (repr->flags & HAS_UNDEFINED_MIN) {
+		*val = scalar_type_min(format);
+	} else if (repr->flags & HAS_UNDEFINED_MAX_MINUS_1) {
+		*val = scalar_type_max(format) - 1;
+	} else {
+		return false;
+	}
+
+	return true;
+}
+
+static bool scalar_invalid_raw(const struct bt_mesh_sensor_format *format,
+			       int64_t *val)
+{
+	const struct scalar_repr *repr = format->user_data;
+
+	if (repr->flags & HAS_INVALID) {
+		*val = scalar_type_max(format) - 1;
+		if (repr->flags & HAS_UNDEFINED_MAX_MINUS_1) {
+			*val += 1;
+		}
+		return true;
+	}
+
+	return false;
+}
+
+static enum bt_mesh_sensor_value_status
+scalar_to_raw(const struct bt_mesh_sensor_value *sensor_val, int64_t *val)
+{
+	if (scalar_decode_raw(sensor_val->format, sensor_val->raw, val) != 0) {
+		return BT_MESH_SENSOR_VALUE_CONVERSION_ERROR;
+	}
+
+	const struct scalar_repr *repr = sensor_val->format->user_data;
```

Comment:
Can user_data ever be NULL? If not disregard this comment, if not do a NULL check.

Reply:
All the scalar type formats are initialized using macros which also sets `user_data`, so no, it won't be NULL